eventmesh-storage-plugin/eventmesh-storage-standalone/src/main/java/org/apache/eventmesh/storage/standalone/broker/MessageQueue.java

```diff
@@ -143,24 +147,30 @@ public MessageEntity getTail() {
     }
 
     /**
-     * Get the message by offset, since the offset is increment, so we can get the first message in this queue and calculate the index of this offset
+     * Retrieves the MessageEntity object with the specified offset.
      *
-     * @param offset
-     * @return MessageEntity
+     * @param offset The offset of the MessageEntity object to be retrieved.
+     * @return The MessageEntity object with the specified offset, or null if no such object exists in the queue.
+     * @throws RuntimeException if the specified offset is less than the offset of the head MessageEntity object.
      */
     public MessageEntity getByOffset(long offset) {
         ReentrantLock lock = this.lock;
         lock.lock();
         try {
-            MessageEntity head = getHead();
-            if (head == null) {
+            if (count == 0) {
                 return null;
             }
+            int headIndex = takeIndex;
+            int tailIndex = putIndex - 1;
+            MessageEntity head = itemAt(headIndex);
             if (head.getOffset() > offset) {
                 throw new RuntimeException(String.format("The message has been deleted, offset: %s", offset));
             }
-            MessageEntity tail = getTail();
-            if (tail == null || tail.getOffset() < offset) {
+            if (tailIndex < 0) {
+                tailIndex += items.length;
+            }
+            MessageEntity tail = itemAt(tailIndex);
+            if (tail.getOffset() < offset) {
```

Comment:
The logic of returning null when `tail == null` is lost in your modification. CC @xwm1992